main/src/main/java/org/mobilitydata/gtfsvalidator/validator/UrlConsistencyValidator.java

```diff
@@ -0,0 +1,244 @@
+  private final GtfsRouteTableContainer routeTable;
+  private final GtfsStopTableContainer stopTable;
+
+  @Inject
+  UrlConsistencyValidator(GtfsAgencyTableContainer agencyTable,
+      GtfsRouteTableContainer routeTable, GtfsStopTableContainer stopTable) {
+    this.agencyTable = agencyTable;
+    this.routeTable = routeTable;
+    this.stopTable = stopTable;
+  }
+
+  @Override
+  public void validate(NoticeContainer noticeContainer) {
+    Map<String, GtfsAgency> agencyByUrlMap = agenciesByUrlMap(agencyTable);
+    for (GtfsRoute route : routeTable.getEntities()) {
+      if (!route.hasRouteUrl()) {
+        return;
+      }
+      GtfsAgency agency = agencyByUrlMap.get(route.routeUrl().toLowerCase());
+      if (agency != null) {
+        noticeContainer.addValidationNotice(
+            new SameRouteAndAgencyUrlNotice(
+                route.csvRowNumber(),
+                route.routeId(),
+                agency.agencyName(),
+                route.routeUrl(),
+                agency.csvRowNumber()
+            ));
+      }
+    }
+    for (GtfsStop stop : stopTable.getEntities()) {
+      if (!stop.hasStopUrl()) {
+        return;
+      }
+      GtfsAgency agency = agencyByUrlMap.get(stop.stopUrl().toLowerCase());
+      if (agency != null) {
+        noticeContainer.addValidationNotice(
+            new SameStopAndAgencyUrlNotice(
+                stop.csvRowNumber(),
+                stop.stopId(),
+                agency.agencyName(),
+                stop.stopUrl(),
+                agency.csvRowNumber()
+            ));
+      }
+    }
+    Map<String, GtfsRoute> routesByUrlMap = routesByUrlMap(routeTable);
+    Map<String, GtfsStop> stopsByUrlMap = stopsByUrlMap(stopTable);
+
+    Maps.filterEntries(stopsByUrlMap,
+        entry -> routesByUrlMap.get(entry.getValue().stopUrl()) != null)
+        .values().forEach(stopWithDuplicateUrl -> {
+      noticeContainer.addValidationNotice(
+          new SameStopAndRouteUrlNotice(
+              stopWithDuplicateUrl.csvRowNumber(),
+              stopWithDuplicateUrl.stopId(),
+              stopWithDuplicateUrl.stopUrl(),
+              routesByUrlMap.get(stopWithDuplicateUrl.stopUrl()).routeId(),
+              routesByUrlMap.get(stopWithDuplicateUrl.stopUrl()).csvRowNumber()
+          ));
+    });
+  }
+
+  /**
+   * Maps {@code GtfsAgency}s by there URLs if provided.
+   *
+   * @param stopTable the {@code GtfsStopTableContainer} to extract {@code GtfsStop} from
+   * @return routes from {@code GtfsStopTableContainer}s mapped by there {@code routes.route_url}
+   * (in lower case) if provided.
+   */
+  private Map<String, GtfsStop> stopsByUrlMap(GtfsStopTableContainer stopTable) {
+    Map<String, GtfsStop> stopsByUrl = new HashMap<>();
+    stopTable.getEntities().forEach(stop -> {
+      if (stop.hasStopUrl()) {
+        if (stopsByUrl.get(stop.stopUrl()) == null) {
+          stopsByUrl.put(stop.stopUrl().toLowerCase(), stop);
+        }
```

Comment:
This looks inconsistent: you call `get()` for the original URL but you `put` a lowercase URL to the map.

Instead, you can just call `putIfAbsent` here and below.